source/extensions/filters/network/zookeeper_proxy/zookeeper_filter.h

```diff
@@ -0,0 +1,116 @@
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ZooKeeperProxy {
+
+/**
+ * All ZooKeeper proxy stats. @see stats_macros.h
+ */
+// clang-format off
+#define ALL_ZOOKEEPER_PROXY_STATS(COUNTER)                              \
+  COUNTER(decoder_error)                                                \
+  COUNTER(connect)                                                      \
+  COUNTER(connect_readonly)                                             \
+  COUNTER(getdata_rq)                                                   \
+  COUNTER(create_rq)                                                    \
+  COUNTER(create2_rq)                                                   \
+  COUNTER(setdata_rq)                                                   \
+  COUNTER(getchildren_rq)                                               \
+  COUNTER(getchildren2_rq)                                              \
+  COUNTER(remove)                                                       \
+  COUNTER(exists)                                                       \
+  COUNTER(getacl)                                                       \
+  COUNTER(setacl)                                                       \
+  COUNTER(sync)                                                         \
+  COUNTER(ping)                                                         \
+  COUNTER(multi)                                                        \
+  COUNTER(create2)                                                      \
+  COUNTER(reconfig)                                                     \
+  COUNTER(createsession)                                                \
+  COUNTER(close)                                                        \
+  COUNTER(setauth)                                                      \
+  COUNTER(setwatches)                                                   \
+// clang-format on
+
+/**
+ * Struct definition for all ZooKeeper proxy stats. @see stats_macros.h
+ */
+struct ZooKeeperProxyStats {
+  ALL_ZOOKEEPER_PROXY_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+/**
+ * Configuration for the ZooKeeper proxy filter.
+ */
+class ZooKeeperFilterConfig {
+public:
+  ZooKeeperFilterConfig(const std::string &stat_prefix, Stats::Scope& scope);
+
+  const ZooKeeperProxyStats& stats() { return stats_; }
+
+  Stats::Scope& scope_;
+  const std::string stat_prefix_;
+  ZooKeeperProxyStats stats_;
+
+private:
+  ZooKeeperProxyStats generateStats(const std::string& prefix,
+                                Stats::Scope& scope) {
+    return ZooKeeperProxyStats{
+        ALL_ZOOKEEPER_PROXY_STATS(POOL_COUNTER_PREFIX(scope, prefix))};
+  }
+};
+
+using ZooKeeperFilterConfigSharedPtr = std::shared_ptr<ZooKeeperFilterConfig>;
+
+/**
+ * Implementation of ZooKeeper proxy filter.
+ */
+class ZooKeeperFilter : public Network::Filter, DecoderCallbacks, Logger::Loggable<Logger::Id::filter> {
+public:
+  ZooKeeperFilter(ZooKeeperFilterConfigSharedPtr config);
```

Comment:
add `explicit`

Reply:
ditto with not happening in other filters, is this new?